virtcontainers/device/drivers/utils.go

```diff
@@ -52,23 +55,17 @@ func isPCIeDevice(bdf string) bool {
 	if len(strings.Split(bdf, ":")) == 2 {
 		bdf = PCIDomain + ":" + bdf
 	}
-	slots, err := ioutil.ReadDir(config.SysBusPciSlotsPath)
+
+	configPath := filepath.Join(config.SysBusPciDevicesPath, bdf, "config")
+	fi, err := os.Stat(configPath)
 	if err != nil {
-		deviceLogger().WithError(err).WithField("path", config.SysBusPciSlotsPath).Warn("failed to list pci slots")
-		return false
-	}
-	b := strings.Split(bdf, ".")[0]
-	for _, slot := range slots {
-		address := getPCISlotProperty(slot.Name(), PCISysFsSlotsAddress)
-		if b == address {
-			maxBusSpeed := getPCISlotProperty(slot.Name(), PCISysFsSlotsMaxBusSpeed)
-			if strings.Contains(maxBusSpeed, PCIeKeyword) {
-				return true
-			}
-		}
+		deviceLogger().WithField("dev-bdf", bdf).WithField("error", err).Warning("Couldn't stat() configuration space file")
```

Comment:
I think you can use `WithError(err)`, instead of `WithField("error", err)`, but that's a nitpick.